spec/requests/api/docs/v1/applications_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+require "swagger_helper"
+
+RSpec.describe "NPQ Applications endpoint", type: :request, openapi_spec: "v1/swagger.yaml" do
+  include_context "with authorization for api doc request"
+
+  let!(:application) { create(:application, lead_provider:) }
+
+  path "api/v1/npq-applications" do
```

Comment:
Is it worth pulling a lot of this repeated logic that's going to be common to all index/show actions into `shared_contexts`? I did this with the request specs for applications/statements here and I think it makes it a lot cleaner/easier to maintain:

https://github.com/DFE-Digital/npq-registration/blob/1747c6826b1634e6338f2c115f09064a87cfd8e9/spec/support/shared_examples/api_show_endpoint_support.rb

https://github.com/DFE-Digital/npq-registration/blob/1747c6826b1634e6338f2c115f09064a87cfd8e9/spec/support/shared_examples/api_index_endpoint_support.rb

The specs then become:

https://github.com/DFE-Digital/npq-registration/blob/1747c6826b1634e6338f2c115f09064a87cfd8e9/spec/requests/api/v3/applications_spec.rb#L9-L33